src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -200,25 +200,33 @@ using StrlenExpression =
 
 // LCASE
 [[maybe_unused]] auto lowercaseImpl =
-    [](std::optional<std::string> input) -> IdOrLiteralOrIri {
+    [](std::optional<ad_utility::triple_component::Literal> input)
+    -> IdOrLiteralOrIri {
   if (!input.has_value()) {
     return Id::makeUndefined();
   } else {
-    return toLiteral(ad_utility::utf8ToLower(input.value()));
+    auto new_content =
+        ad_utility::utf8ToLower(asStringViewUnsafe(input.value().getContent()));
+    input.value().replaceContentWithSameLength(new_content);
```

Comment:
As the uppercasing and lowercasing depends on the language and subtleties of UTF-8, I don't think it is guaranteed to have the same lenght, and you also don't need this.
Just implement an efficient `replaceContent` function that always does the most efficient thing possible for all cases (shorter, longer, same length).